pkg/adapter/containers_remote.go

```diff
@@ -464,19 +464,33 @@ func (r *LocalRuntime) Run(ctx context.Context, c *cliconfig.RunValues, exitCode
 	results := shared.NewIntermediateLayer(&c.PodmanCommand, true)
 	cid, err := iopodman.CreateContainer().Call(r.Conn, results.MakeVarlink())
 	if err != nil {
-		return 0, err
+		return exitCode, err
 	}
 	if c.Bool("detach") {
 		_, err := iopodman.StartContainer().Call(r.Conn, cid)
 		fmt.Println(cid)
-		return 0, err
+		return exitCode, err
 	}
 	errChan, err := r.attach(ctx, os.Stdin, os.Stdout, cid, true, c.String("detach-keys"))
 	if err != nil {
-		return 0, err
+		return exitCode, err
+	}
+	err = <-errChan
+	if err != nil {
+		return exitCode, err
+	}
+	if ecode, err := iopodman.WaitContainer().Call(r.Conn, cid, 0); err == nil {
+
+		exitCode = int(ecode)
```

Comment:
Is it possible for the container to have been removed right before the call to `iopodman.WaitContainer().Call(r.Conn, cid, 0)` which (according to the function comment) will return a `ContainerNotFound` error.  However, below in the `rm` condition, I don't see this being ignored (removal was the desired outcome) -  In other words, I fear the condition below would throw a useless removal error.  Though my golang-foo is weak, so maybe I'm missing something.

I can't see how this would lead to a hang though...as I'm seeing in my manual testing :confounded: